x-pack/plugins/reporting/server/core.ts

```diff
@@ -235,6 +224,30 @@ export class ReportingCore {
     this.pluginSetup$.next(true);
   }
 
+  /**
+   * Validate export types with config settings
+   * only CSV export types should be registered in the export types registry for serverless
+   */
+  private validateExportTypesWithConfig(): ExportType[] {
+    if (!this.config.export_types.pdf.enabled || !this.config.export_types.png.enabled) {
+      this.exportTypes.push(
+        new CsvSearchSourceExportType(this.core, this.config, this.logger, this.context)
+      );
+      this.exportTypes.push(new CsvV2ExportType(this.core, this.config, this.logger, this.context));
+    } else {
+      this.exportTypes.push(
+        new CsvSearchSourceExportType(this.core, this.config, this.logger, this.context)
+      );
+      this.exportTypes.push(new CsvV2ExportType(this.core, this.config, this.logger, this.context));
+      this.exportTypes.push(new PdfExportType(this.core, this.config, this.logger, this.context));
+      this.exportTypes.push(new PngExportType(this.core, this.config, this.logger, this.context));
+      // deprecated export types for tests
+      this.exportTypes.push(new PdfV1ExportType(this.core, this.config, this.logger, this.context));
+      this.exportTypes.push(new PngV1ExportType(this.core, this.config, this.logger, this.context));
+    }
+    return this.exportTypes;
```

Comment:
I think we can remove `this.exportTypes` in this PR, and simplify the workings of the core class by taking out some state that is not needed after startup is done.

let's remove this line:
```
  113   private exportTypes: ExportType[] = [];
```

calling this this function and registering the export types can look like: 
```typescript
    this.validateExportTypesWithConfig().forEach((et) => { // or rename the private helper to this.getExportTypes
      this.exportTypesRegistry.register(et);
    });
```

calling `setup` and `start` on the export types can look like:
```
    this.exportTypesRegistry.getAll().forEach((et) => {
      et.setup(setupDeps);
    });
```

This function then doesn't need to update the `this` context. It would just create an empty array, build up the array in the body, and return the array.

Reply:
Thank you! [6d4273b](https://github.com/elastic/kibana/pull/162815/commits/6d4273b8b051db14044c51d821bea2bba69af627)